src/mod_muc_sql.erl

```diff
@@ -236,6 +238,27 @@ set_nick(LServer, Host, From, Nick) ->
 	end,
     ejabberd_sql:sql_transaction(LServer, F).
 
+-spec decode_affiliation(Arg :: binary()) -> atom().
```

Comment:
👏

Reply:
Pattern matching ftw, you can probably use something similar instead of the atom_to_binary stuff (its safer even though we're pretty sure this is innocuous code)

Reply:
and constant time, even though I'm sure the c code underneath is probably